src/nutrients_plankton_bacteria_detritus.jl

```diff
@@ -85,70 +97,106 @@ end
 #   -> P⁺ = P⁻ / (1 + Δt * m)
 #
 
-@inline bacteria_production(μᵇ, kᴰ, D, B) = μᵇ * D / (D + kᴰ) * B
-@inline plankton_production(μᵖ, kᴺ, kᴵ, I, N, P) = μᵖ * (N / (N + kᴺ) * I / (I + kᴵ)) * P
-@inline bacteria_mortality(m, B) = m * B^2
-@inline plankton_mortality(m, P) = m * P^2
+# A depth-dependent temperature curve from Zakem (2018)
+# Temp = 12 .*exp.(z./ 150) .+ 12 .*exp.(z ./ 500) .+ 2
+
+# Temperature modification to metabolic rates, following the Arrhenius equation
+# @inline temp_fun(Temp) = 0.8 .* exp.(-4000 .*(1 ./ (Temp .+ 273.15) .- 1 ./ 293.15))
+
+@inline bacteria_production(μᵇ, kᴰ, D, B) = (μᵇ * D / (D + kᴰ) * B) 
+@inline phytoplankton_production(μᵖ, kᴺ, kᴵ, I, N, P) = (μᵖ * min(N / (N + kᴺ) , I / (I + kᴵ)) * P) 
+@inline zooplankton_production(gₘ, kᴾ, γ, P, Z) = γ * (gₘ * P / (P + kᴾ) * Z) 
+@inline bacteria_mortality(mlin,mq, B) = mlin * B + mq * B^2
+@inline phytoplankton_mortality(mlin,mq, P) = mlin * P + mq * P^2
+@inline zooplankton_mortality(mlin,mq, Z) = mlin * Z + mq * Z^2
 
 @inline function (bgc::NutrientsPlanktonBacteriaDetritus)(i, j, k, grid, ::Val{:N}, clock, fields)
     μᵖ = bgc.maximum_plankton_growth_rate
     μᵇ = bgc.maximum_bacteria_growth_rate
+    gₘ = bgc.maximum_grazing_rate
     kᴰ = bgc.detritus_half_saturation
     kᴺ = bgc.nutrient_half_saturation
+    kᴾ = bgc.phytoplankton_half_saturation
     kᴵ = bgc.PAR_half_saturation
     λ = bgc.PAR_attenuation_scale
-    m = bgc.quadratic_mortality_rate
     y = bgc.bacteria_yield
+    γ = bgc.zooplankton_assimilation
```

Comment:
I know this is historical and exactly how I have it in my own models, so Hengdi it was the right thing to do to follow this convention, but now I am wondering in our "revisit" here if we might just also call it a yield, to be really consistent. (Previous models that used gamma for Z did not have any parameter called yield.) What do you think? Should we have a yield array, as long as the number of functional types? Or, one yield for each type?

Reply:
eg this would be `zooplankton_yield`?

Sounds like you may have some nice ideas for generalizing this model to be somethiing truly amazing, so perhaps comment is out of date

Reply:
Hengdi and I just met in person and went through all of these things, so I'm going to mark a lot of them as resolved!